src/arith/rewrite_simplify.cc

```diff
@@ -697,7 +698,7 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const SubNode* op) {
     TVM_TRY_RECURSIVE_REWRITE(x - (y + c1), (x - y) + (0 - c1));
     TVM_TRY_RECURSIVE_REWRITE(x - (y - z), (x + z) - y);
     TVM_TRY_RECURSIVE_REWRITE(x - y * c1, x + y * (0 - c1));
-  } else if (op->dtype.is_float()) {
+  } else {
```

Comment:
For currently unsupported integer types, is there any reason to take this path rather than path for the index types? They are also integers and this path has more rules defined. But, I'm not currently sure if it supports negative index but if it does not, it could be effectively same with unsigned integer?

Reply:
Some of the rewrite rules in the `IsIndexDType` branch assume that overflow won't occur.  While this isn't guaranteed to be true, we're more likely to get overflow for small integer types than for the 32/64-bit signed integers used for index dtypes.

For the uint16/uint8 types seen in https://github.com/apache/tvm/pull/17149, it would be a lot easier to get accidental overflow.